From 855a133d55077c47f8cf5e7b75751cb5e2c376a9 Mon Sep 17 00:00:00 2001
From: Jami Kettunen <jami.kettunen@protonmail.com>
Date: Fri, 27 May 2022 19:33:35 +0300
Subject: [PATCH 3/4] CMake: Make running tests optional

Also makes dbustest-1 an optional dependency with -DENABLE_TESTS=OFF.
---
 CMakeLists.txt | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index dbda8a8..f32e13d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,7 +9,7 @@ else()
   option (werror "Treat warnings as errors." TRUE)
 endif()
 
-enable_testing ()
+option(ENABLE_TESTS "Enable all tests and checks" ON)
 option (ENABLE_COVERAGE "Generate Coverage Reports" ON)
 
 set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake" "${CMAKE_MODULE_PATH}")
@@ -66,8 +66,10 @@ include_directories(${GIO_UNIX2_INCLUDE_DIRS})
 pkg_check_modules(JSONGLIB REQUIRED json-glib-1.0)
 include_directories(${JSONGLIB_INCLUDE_DIRS})
 
-pkg_check_modules(DBUSTEST REQUIRED dbustest-1>=14.04.0)
-include_directories(${DBUSTEST_INCLUDE_DIRS})
+if(${ENABLE_TESTS})
+  pkg_check_modules(DBUSTEST REQUIRED dbustest-1>=14.04.0)
+  include_directories(${DBUSTEST_INCLUDE_DIRS})
+endif()
 
 pkg_check_modules(SQLITE REQUIRED sqlite3)
 include_directories(${SQLITE_INCLUDE_DIRS})
@@ -104,7 +106,10 @@ add_subdirectory(liblomiri-url-dispatcher)
 add_subdirectory(tools)
 add_subdirectory(gui)
 add_subdirectory(po)
-add_subdirectory(tests)
+if(${ENABLE_TESTS})
+  enable_testing()
+  add_subdirectory(tests)
+endif()
 
 # coverage reporting
 if (${ENABLE_COVERAGE})
-- 
2.36.1

